### DISCUSSION TITLE
Next.js 15's ISR behaves differently from Next.js 14's ISR

### DISCUSSION THREAD

**Discussion by lid0a** [vercel/next.js#72965] [App Router] [answered]
Here are my two applications with the same code: [the first is created using next.js 14](https://codesandbox.io/p/devbox/9fjjhc) and [the second uses next.js 15](https://codesandbox.io/p/devbox/3wqzcw). Both versions use time-based ISR. Version 14 regenerates the page even there was a 404 error whereas version 15 shows 404 page permanently. Is that intended behaviour or a bug? I couldn't find any information about this on the docs. 

To reproduce, you should run the applications in production mode (`pnpm build && pnpm start`).

**Comment by astrodomas** [comment]
```
export const revalidate = 10;
export const dynamicParams = true;
export const dynamic = 'force-static';
export async function generateStaticParams() {
  return [];
}
```
Should work as expected

**Reply by lid0a** [reply, marked as answer]
Hi! Sorry for late response. Unfortunately, the suggested code didn't change anything. But it seems that the bug is now fixed in the latest version of nextjs (15.1.6). Btw, `revalidate` config option is enough in this case because it is a static page. No need to use `dynamicParams`, `dynamic`, `generateStaticParams`.
